src/components/Preferences/LocalIdentities.js

```diff
@@ -19,91 +20,149 @@ import {
 } from '../IdentityManager/IdentityManager'
 import EmptyLocalIdentities from './EmptyLocalIdentities'
 import Import from './Import'
+import { GU } from '../../utils'
 
 const iOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream
 
-const LocalIdentities = ({
-  dao,
-  localIdentities,
-  onClearAll,
-  onImport,
-  onModify,
-  onModifyEvent,
-}) => {
-  // transform localIdentities from object into array and attach address to each entry
-  const identities = Object.entries(localIdentities).map(
-    ([address, identity]) => ({
-      ...identity,
-      address,
+const LocalIdentities = React.memo(
+  ({ dao, localIdentities, onClearAll, onImport, onModify, onModifyEvent }) => {
+    if (!Object.keys(localIdentities).length) {
+      return <EmptyLocalIdentities onImport={onImport} />
+    }
+
+    const { identityEvents$ } = React.useContext(IdentityContext)
+    const { showLocalIdentityModal } = React.useContext(
+      LocalIdentityModalContext
+    )
+    const [addressesSelected, setAddressesSelected] = React.useState(
+      new WeakMap()
+    )
+    React.useEffect(() => {
+      const newAddressesSelected = new WeakMap()
+      Object.keys(localIdentities).forEach(key => {
+        newAddressesSelected.set(localIdentities[key], true)
+      })
+      setAddressesSelected(newAddressesSelected)
+    }, [localIdentities])
```

Comment:
I wonder if the re-rendering was happening because we weren't using a `useEffect()` before, rather than the `Map`?